dash/dash.py

```diff
@@ -492,7 +492,8 @@ def serve_component_suites(self, package_name, path_in_package_dist):
 
         mimetype = ({
             'js': 'application/JavaScript',
-            'css': 'text/css'
+            'css': 'text/css',
+            'map': 'application/json'
         })[path_in_package_dist.split('.')[-1]]
```

Comment:
I don't like this split, if someone send a filename with an extension that is not in the dict it will throw. We can go around that by using `.get` instead and return a 400 instead of a 500 if the content_type is None. But I would still prefer if we used `mimetypes.guess_type`.

Reply:
Same comment as above. This is to get base source map functionality in. This brings other concerns in that should be able to stand on their own and be their own issue or PR and feels out of scope.

Reply:
What I'm saying here is before adding the map option it would raise an error on requesting the map because of the direct access. We should put the dict in a variable and add a default: 

```python
content_types = {
    'map': 'application/json',
    ...
}
content_type = content_types.get(path_in_package_dist, 'text/plain')
```

I don't think guessing the content types with `mimetypes` would be enough to solve #96 anyway, the issue is more complex than that.

Reply:
Problem with setting a default is that it prevents content type sniffing from browsers capable of doing so -- the relevant entry from the norm:

https://tools.ietf.org/html/rfc7231#section-3.1.1.5
> If a Content-Type header field is not present, the recipient
   MAY either assume a media type of "application/octet-stream"
   [...] or examine the data to determine its type.

Better off not applying defaults and letting the client do its thing if we do not have an opinion.

If we want a fast way to support unknown file types without providing the content-type, we're better off letting it pass through and just not providing a content type. As you pointed above for the ```mimetypes``` the issue issue is more complex than that.

I will create an issue for general file type support and include both the norm entry here and the ```mimetypes``` package solution in it. A combination of both is probably what we need: Mimetypes for 90%+ of cases, overrides for known cases (.map is not in the list), pass through / no content-type for unknown types.

Reply:
Yes, what I want is the content type sniffing from the browser, so no default but still a `.get` so it doesn't throw. Thanks for doing the research.

Reply:
I don't think this has to be changed here, that is outside of the scope of what this PR does -- just adding the `map` type by using the same pattern we have used for `js` and `css`, and that old pattern would throw an error. If we want to change the pattern that should be done in a different PR, perhaps in the same one we add the `mimetypes` module.